hack/jenkins/e2e.sh

```diff
@@ -913,6 +913,7 @@ case ${JOB_NAME} in
     NUM_NODES="10"
     MASTER_SIZE="n1-standard-2"
     NODE_SIZE="n1-standard-1"
+    E2E_ZONE="asia-east1-a"
```

Comment:
`kubernetes-kubemark-gce` currently sets this in Jenkins config. For this PR, I've moved it out and into the case statement, and it only hits this job. Other jobs will remain where they are.

I'd be happy to change this if this job should be running elsewhere.  `kubernetes-e2e-gce-enormous-cluster` also runs in `asia-east1-a`.